service/controller/v18/resource/cloudformation/current.go

```diff
@@ -56,8 +56,22 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found stack %#q in the AWS API", stackName))
 				r.logger.LogCtx(ctx, "level", "debug", "message", "keeping finalizer")
 				finalizerskeptcontext.SetKept(ctx)
+
+				break
 			}
 		}
+
+		// When a tenant cluster is deleted it might be not completely created yet
+		// in the first place. There can be issues with unaccessible stack output
+		// values in such cases, causing the deletion process to get into a
+		// deadlock. To remedy such cases we simply return the stack state
+		// containing the stack name, without trying to access any stack output
+		// values.
+		currentState := StackState{
+			Name: key.MainGuestStackName(customObject),
```

Comment:
This seems like a good idea, I am just not sure if our resources execute fine if only the name is set.

Reply:
They do, except some error handling. Which I am fixing still. Stay tuned.